### PR TITLE
Inactivity probe for ovndb connection

### PR DESCRIPTION
**Pull request by pperiyasamy**
This enables inactivity probe on the ovn's nbdb and sbdb connections which helps to reconnect with ovndb immediately
when there is an disconnect happened accidentally with database.

### CODE REVIEW

**Comment by tssurya** [comment]
some questions on the overall logic I have not yet dived deeply into the PR

> ho request periodically at 5 sec interval to ovn nbdb and sbdb connections, when echo request fails consecutively 2 times consider connection as inactive. When ovn dbs is back online, then reconnect with ovndb. It ensures early detection of ovndb connectivity issues and connect back immediately when it's back to operational.

how does this fair with what we do today? you mention this will help us detect connection issues earlier than we do today, trying to understand by how much and does this goroutine have a perf impact?

**Comment by pperiyasamy** [comment]
> how does this fair with what we do today? you mention this will help us detect connection issues earlier than we do today, trying to understand by how much and does this goroutine have a perf impact?

@tssurya Yes, there is a [reconnect](https://github.com/ovn-org/libovsdb/blob/main/client/client.go#L1176) loop exists in client upon close of disconnect channel. But it seems like that takes hours to close of disconnect channel, hence the delay with reconnect. that was my understanding. with Echo, inactivity would have figured out sooner. @dcbw would have more understanding of this problem.
this echo go routine runs at 5 sec interval and not in a busy loop. so it may not have a perf impact.

**Comment by pperiyasamy** [comment]
@tssurya The probe check is moved into libovsdb with another [PR](https://github.com/ovn-org/libovsdb/pull/356), will adapt this ovn-kubernetes PR `WithInactivityCheck` option once libovsdb PR is accepted/merged.

**Comment by tssurya** [comment]
ACK I can wait for the libovsdb changes to land, maybe @dcbw or @jcaamano know this better - but 
```
But it seems like that takes hours to close of disconnect channel, hence the delay with reconnect. that was my understanding.
```
why is that? Shouldn't we try and fix that close taking hours to happen first? Rather than adding a new healthcheck to circumvent that?
I am in favour of "failing-fast" by all means, but trying to see why we need a whole new routine doing this every 5 seconds if we can fix the root cause? (I'll paste the same question on the libovsdb PR - again not saying we don't need this, just trying to understand why part)

File: go-controller/pkg/ovn/copp_test.go
Comment (jcaamano) [inline go-controller/pkg/ovn/copp_test.go:131]: hmmm can't wrap my head around this one.

how can there multiple empty-name default COPPs in a real DB? 

I guess the name index was added in an schema upgrade, but wouldn't the DB be forced to handle that in some way before we even notice it client-side?

Or is this a test targeting an scenario where we were mopping up the DB before that schema upgrade actually happened?

**Review by jcaamano — APPROVED** [review]

File: go-controller/pkg/ovn/copp_test.go
Comment (dcbw) [reply go-controller/pkg/ovn/copp_test.go:131]: @jcaamano Yes, that was the issue. Before OVN 22.09 COPP objects had no name and no index on name. So here we're testing that we can fix the DB on upgrade.

**Pull request merged**
